drivers/timer/arm_arch_timer.c

```diff
@@ -58,7 +66,7 @@ static void arm_arch_timer_compare_isr(const void *arg)
 
 	uint64_t curr_cycle = arm_arch_timer_count();
 	uint64_t delta_cycles = curr_cycle - last_cycle;
-	uint32_t delta_ticks = (cycle_diff_t)delta_cycles / CYC_PER_TICK;
+	uint32_t delta_ticks = TO_CYCLE_DIFF(delta_cycles) / CYC_PER_TICK;
```

Comment:
What about removing the `TO_CYCLE_DIFF`, just using `uint32_t delta_ticks = (cycle_diff_t) (delta_cycles / CYC_PER_TICK);`

Reply:
Then, any differences between `uint32_t delta_ticks = delta_cycles / CYC_PER_TICK` and `uint32_t delta_ticks = (uint32_t) delta_cycles / CYC_PER_TICK` as `delta_ticks` is a `uint32_t`

Reply:
Oh, sorry, you are right. I didn't notice that the parenthesis encloses the whole division in your previous comment.

I would still keep the macro, because we want to cast the `delta_cycles` variable to `uint32_t` if possible to improve performance.

Reply:
Okay, fair enough